apis/secretmanager/v1beta1/secret_reference.go

```diff
@@ -0,0 +1,181 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1beta1
+
+import (
+	"context"
+	"fmt"
+	"strings"
+
+	refsv1beta1 "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/k8s"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	"k8s.io/apimachinery/pkg/apis/meta/v1/unstructured"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+var _ refsv1beta1.ExternalNormalizer = &SecretManagerSecretRef{}
+
+// SecretManagerSecretRef defines the resource reference to SecretManagerSecret, which "External" field
+// holds the GCP identifier for the KRM object.
+type SecretManagerSecretRef struct {
+	// A reference to an externally managed SecretManagerSecret resource.
+	// Should be in the format "projects/<projectID>/locations/<location>/secrets/<secretID>".
+	External string `json:"external,omitempty"`
+
+	// The name of a SecretManagerSecret resource.
+	Name string `json:"name,omitempty"`
+
+	// The namespace of a SecretManagerSecret resource.
+	Namespace string `json:"namespace,omitempty"`
+
+	parent *SecretManagerSecretParent
```

Comment:
nit: suggest adding a comment saying that this field is optional. Some resources does not necessary to have `parent` field.

Reply:
do you mean adding the comment in the code template? We can do that, but I would assume the contributors can figure it out themselves.